include/oneapi/dpl/pstl/algorithm_impl.h

```diff
@@ -340,7 +340,7 @@ __pattern_walk2(_ExecutionPolicy&& __exec, _ForwardIterator1 __first1, _ForwardI
         typedef typename ::std::iterator_traits<_ForwardIterator2>::reference _ReferenceType2;
 
         __par_backend::__parallel_for_each(::std::forward<_ExecutionPolicy>(__exec), __begin, __end,
-                                           [&__f](const ::std::tuple<_ReferenceType1, _ReferenceType2>& __val) {
+                                           [&__f](::std::tuple<_ReferenceType1, _ReferenceType2> __val) {
```

Comment:
@MikeDvorskiy 

For my edification, why are we changing this parameter type to pass the tuple by value rather than const lvalue ref?

Is it required for compilation or merely a better way of handling zip iterator references?

I understand that the zip iterator reference is a tuple of references, but I'm not immediately seeing the advantage of using the tuple passed by value over the lvalue reference to a tuple.

Reply:
Short answer - it doesn't compile with `const tuple&`.

Longer answer - if `tuple` is `const` it preserves constness for its objects. So when you have `zip_forward_iterator<..., zip_iterator>` and dereference it and bind it to `const&` you have `const tuple<_whatever_refs_, tuple<_refs_>>&`. Then, you try to read inner `tuple` from outer `tuple` and the only thing you can get is the `const tuple<_refs_>&` (unless you do copy but this is the different story). But this is `transform` algorithm and eventually you are trying to write to `const tuple<_refs_>&` which is obviously compile-time error.

For even longer answer, don't hesitate to reach me personally :)

Reply:
Alright, I think I understand. This element is from an outer zip iterator, which zips together both inputs and outputs on the way to the functor.  Works OK with normal lvalue references as output, but when a zip iterator is present as the output, you end up with a const where you don't want one. 

I was confused because I was conflating the inner and outer zip iterators.
Thanks!

Reply:
About `const` I understood and agree, thank you @rarutyun.
But why we remove reference `&` too?

Reply:
We removed the reference too, because a tuple argument passed as rvalue  doesn't  match to lvalue (tuple<..>&).

Reply:
ok